pkg/provider/common.go

```diff
@@ -31,3 +34,35 @@ func buildAPIReqAssignTags(tfRsrcTags []commonTerraform.Tag) []commonApi.Tag {
 	}
 	return tags
 }
+
+var resourceWal = schema.SingleNestedAttribute{
+	Description: "Write-Ahead Logs (WAL) Storage.",
+	Optional:    true,
+	Attributes: map[string]schema.Attribute{
+		"iops": schema.StringAttribute{
+			Description:   "IOPS for the selected volume. It can be set to different values depending on your volume type and properties.",
+			Optional:      true,
+			Computed:      true,
+			PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+		},
+		"size": schema.StringAttribute{
+			Description:   "Size of the volume. It can be set to different values depending on your volume type and properties.",
+			Required:      true,
```

Comment:
size is actually a mandatory field. In the request if I just put P1 and leave size as nil it will return with this:

> biganimal_cluster.single_node_cluster: Creating...
╷
│ Error: status: 400 - Bad Request
│
│   with biganimal_cluster.single_node_cluster,
│   on resource.tf line 21, in resource "biganimal_cluster" "single_node_cluster":
│   21: resource "biganimal_cluster" "single_node_cluster" {
│
│ must have required property 'clusterType'
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties
│ must NOT have additional properties

however if I put P1 and give a size of 7 Gi I get the error:

> biganimal_cluster.single_node_cluster: Creating...
biganimal_pgd.pgd_cluster: Destruction complete after 5s
╷
│ Error: status: 400 - Bad Request
│
│   with biganimal_cluster.single_node_cluster,
│   on resource.tf line 21, in resource "biganimal_cluster" "single_node_cluster":
│   21: resource "biganimal_cluster" "single_node_cluster" {
│
│ Value of storage size must be between 4 Gi and 4 Gi

I guess a future improvement would be to fetch the list of azurepremiumstorage volume properties and their sizes so size can be optional if they select azurepremiumstorage. But I think we can have that as an enhancement

Reply:
volume_properties: P1
volume_type: P1
size: 4  (exact size associated with each volume_property)

so user need to fill above details, for it to work

Reply:
let provide link to volume storage ref in examples